src/test/ts/core/NpmTagsTest.ts

```diff
@@ -32,7 +32,12 @@ describe('NpmTags', () => {
       it('replaces multiple slashes', () =>
         checkSimple('feature/BLAH/12/3', BranchState.Feature, '100.7.22', [ 'feature-BLAH-12-3' ])
       );
+    });
 
+    describe('dependabot branches', () => {
+      it('uses the branch name without npm_and_yarn component', () =>
```

Comment:
Could we add a test here with a really weird and unexpected (imaging they totally changed their conventions)`dependabot` branch name to assert it doesn't fall apart?

Reply:
There are already tests for varying forms of odd branch name handling in this file, so I really don't see what this would do. Also, any test we add for it isn't going to give us any real protection so this is going to provide false hopes.

Reply:
Nonetheless, I've added some additional property tests to test some more randomization than we currently have. They aren't great, but hopefully this will make you feel more reassured.

I've also made it handle any potential dependabot variant and added tests for that.